### PR TITLE
Allow using custom accessControl (eg. OPA)

### PR DESCRIPTION
**Pull request by dantonbertuol**
This PR is the conclusion of [this PR](https://github.com/trinodb/charts/pull/88).

Adds the ability to use custom access control (eg. Open Policy Agent).

### CODE REVIEW

File: charts/trino/values.yaml [hunk 1]
```diff
@@ -90,8 +90,18 @@ server:
 accessControl: {}
 # accessControl -- [System access
 # control](https://trino.io/docs/current/security/built-in-system-access-control.html)
-# configuration.
+# configuration.<br>
+# `type: configmap | custom`<br>
```
Comment (nineinchnick) [inline charts/trino/values.yaml:94]: Remove this line, or change this into a proper English sentence, like so:
> Set the `type` property to either `configmap` and provide the rule file contents in `rules`, or `custom` and provide configuration properties in `properties`.

The examples should be sufficient in explaining how to set the type.

File: charts/trino/templates/configmap-coordinator.yaml [hunk 2]
```diff
@@ -80,14 +80,20 @@ data:
...
+  {{- if eq .Values.accessControl.type "configmap" }}
   access-control.properties: |
     access-control.name=file
     {{- if .Values.accessControl.refreshPeriod }}
     security.refresh-period={{ .Values.accessControl.refreshPeriod }}
     {{- end }}
     security.config-file={{ .Values.server.config.path }}/access-control/{{ .Values.accessControl.configFile | default "rules.json" }}
-{{- end }}{{- end }}
+  {{- end }}
+  {{- if eq .Values.accessControl.type "custom" }}
```
Comment (nineinchnick) [inline charts/trino/templates/configmap-coordinator.yaml:92]: Can you add a condition at the end to check if the type is set to anything other than `configmap` or `custom`, and call the `fail` template function with an actionable error message?

Ideally, we should also check if properties from different types are not mixed. For example, if `customValues` is _not_ set when `type` is `configmap`.

This kind of validation would require adding some more tests, but we don't have any negative tests in `test.sh` right now, so that's optional.

File: test-values.yaml [hunk 3]
```diff
@@ -103,45 +103,48 @@ initContainers:
...
-          {
-            "user": "alice",
-            "owner": false
-          }
-        ]
-      }
+  type: custom
+  customValues: |
+    access-control.name=read-only
+  # type: configmap
```
Comment (nineinchnick) [inline test-values.yaml:109]: Remove commented out code in tests. Add a separate test in `test.sh` so we can cover both cases.

File: charts/trino/README.md [hunk 4]
```diff
@@ -85,7 +85,14 @@ Fast distributed SQL query engine for big data analytics that helps you explore
   ```
 * `accessControl` - object, default: `{}`  
 
-  [System access control](https://trino.io/docs/current/security/built-in-system-access-control.html) configuration.
+  [System access control](https://trino.io/docs/current/security/built-in-system-access-control.html) configuration.<br> `type: configmap | custom`<br> Custom Values Example:
+  ```yaml
+  type: custom
+  customValues: |
```
Comment (nineinchnick) [inline charts/trino/README.md:91]: How about this?
```suggestion
  type: properties
  properties: |
```

**Review by nineinchnick — APPROVED** [review]

**Comment by dantonbertuol** [comment]
Any predictions for the release of this?

**Comment by nineinchnick** [comment]
I'll release a new version once Trino 455 comes out. But I can't tell when that will happen. It's best to ask on the Trino Slack.

**Comment by nineinchnick** [comment]
0.28.0 has just been released
